### DISCUSSION TITLE
Can HiFiC compression be used on multiple GPUs?

### DISCUSSION THREAD

**Discussion by gregg-p** [tensorflow/compression#162] [Q&A]
Is the HiFiC compression/decompression (tfci.py) able to utilize multiple GPUs when running? I have 8 GPUs, but noticed it is currently only using 1. Using all 8 would provide a significant speed increase. Any help is appreciated. Thanks.

**Comment by gtoderici** [comment]
Hi there,

If you want to use more GPUs you need to write code to achieve that.
There's no explicit device placement so you should be able to do that, but
it's unclear what you'd achieve with more GPUs.

George

**Comment by gregg-p** [comment]
**os.environ["CUDA_VISIBLE_DEVICES"] = ""** allows you to specify which CUDA device to run the script on.
At the top of tfci.py, I add: **os.environ["CUDA_VISIBLE_DEVICES"] = "0, 1, 2, 3, 4, 5, 6, 7"**, it recognizes all 8 GPUs, but it still only uses GPU 0. Would modifications need to be made to how the actual model/code functions to utilize more than 1 GPU?
I've noticed this is also the same scenario when training the untrained models.
I appreciate your help & guidance thus far.

**Comment by gtoderici** [comment]
I think you'll need to write code to do the distribution between GPUs. In
particular you'd probably want to load a batch on each GPU explicitly
(i.e., use with tf.device("/gpu:X") for X in 0->7 and only then execute the
graph).

George

**Comment by gpatti-sabre** [comment]
Would using TensorFlow's tf.distribute.MirroredStrategy() be a feasible option for solving this issue?
